### PR TITLE
Introduce bufferization pass

### PR DESCRIPTION
**Pull request by chelini**

### CODE REVIEW

**Review by rengolin — APPROVED** [review]

File: lib/TPP/DefaultTppPasses.cpp [hunk 1]
```diff
@@ -91,18 +91,7 @@ struct DefaultTppPasses : public DefaultTppPassesBase<DefaultTppPasses> {
...
-    buffOpts.allowReturnAllocs = true;
-    buffOpts.bufferizeFunctionBoundaries = true;
-    buffOpts.functionBoundaryTypeConversion =
-        bufferization::LayoutMapOption::IdentityLayoutMap;
-    pm.addPass(bufferization::createOneShotBufferizePass(buffOpts));
-    pm.addPass(bufferization::createDropEquivalentBufferResultsPass());
-    pm.addNestedPass<func::FuncOp>(
-        bufferization::createFinalizingBufferizePass());
-    // Clean up after bufferization.
-    pm.addPass(bufferization::createBufferDeallocationPass());
```
Comment (adam-smnk) [inline lib/TPP/DefaultTppPasses.cpp:104]: This should be added either to the bufferization pass or to the default pipeline after bufferization.

Comment (chelini) [reply lib/TPP/DefaultTppPasses.cpp:104]: is the post-processing in Bufferize.cpp

Comment (adam-smnk) [reply lib/TPP/DefaultTppPasses.cpp:104]: Probably the default pipeline. But tests seems happy so I can add it separately with some test case.

Comment (adam-smnk) [reply lib/TPP/DefaultTppPasses.cpp:104]: Nevermind, I'm just blind. Another `createBufferDeallocationPass` is already present at the end of the default pipeline and there is a test case in `DefaultPipeline/mixed.mlir` that verifies buffer deallocation.
No need for any changes.

**Pull request merged**